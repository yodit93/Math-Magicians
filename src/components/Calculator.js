import { useState } from 'react';
import Result from './Result';
import Input from './Input';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: '0', next: null, operation: null });
  const onHandleClick = (e) => {
    const targetText = e.target.textContent;
    const newState = calculate(state, targetText);
    setState(newState);
  };
  return (
    <div className="math-rapper">
      <div className="math-text">Let&apos;s do some Math</div>
      <div className="calculator">
        <div className="math">
          <Result state={state} />
          <Input onHandleClick={onHandleClick} />
        </div>
      </div>
    </div>
  );
};
export default Calculator;
