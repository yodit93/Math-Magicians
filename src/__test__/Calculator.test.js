import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator componenet', () => {
  test('should render the calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have div with the text "Let&apos;s do some Math" ', () => {
    render(<Calculator />);
    const divElement = screen.getByText("Let's do some Math");
    expect(divElement).toBeInTheDocument();
  });
});
