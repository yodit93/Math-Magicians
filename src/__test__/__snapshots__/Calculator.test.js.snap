// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator componenet should render the calculator component 1`] = `
<div
  className="math-rapper"
>
  <div
    className="math-text"
  >
    Let's do some Math
  </div>
  <div
    className="calculator"
  >
    <div
      className="math"
    >
      <div
        className="result-cont"
        data-testid="test"
      >
        <span
          className="result"
        >
          0
           
           
        </span>
      </div>
      <div
        className="grid"
      >
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="item division color"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="item color"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="item color"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="item color"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="item item0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="item"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="item color"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
